src/main/scala/firrtl/transforms/BlackBoxSourceHelper.scala

```diff
@@ -126,7 +142,7 @@ object BlackBoxSourceHelper {
   def copyResourceToFile(name: String, file: File) {
     val in = getClass.getResourceAsStream(name)
     if (in == null) {
-      throw new FileNotFoundException(s"Resource '$name'")
+      throw new FileNotFoundException(s"$name (No such file or directory)")
```

Comment:
Why not wrap this in `safeFile` or otherwise throw a `BlackBoxNotFoundException`?

Reply:
I'm trying to provide equivalence with how `new FileInputStream(...)` behaves in terms of exceptions thrown so all the `safeFile` wrappers are one level up. Does that make sense?

I could extend `safeFile` to also catch `NullPointerException` and then wrap the file write code in this. However, that seems overly broad. What do you think?

Reply:
Actually, that isn't overly broad as `safeFile` is private... It should be fine to catch the `NullPointerException`. That makes things cleaner. Good suggestion.

Reply:
Fixed in the fixup commit. Catching the `NullPointerException` does mean that the full stack trace won't give you the `FileNotFoundException` below the `BlackBoxNotFoundException`, but I don't think that's a big deal. The top-level exception should be the one that matters and that contains the correct information.